scipy/fft/_pocketfft/tests/test_real_transforms.py

```diff
@@ -0,0 +1,815 @@
+from __future__ import division, print_function, absolute_import
+
+from os.path import join, dirname
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal, assert_equal
+import pytest
+from pytest import raises as assert_raises
+
+from scipy.fft._pocketfft.realtransforms import (
+    dct, idct, dst, idst, dctn, idctn, dstn, idstn)
+
+fftpack_test_dir = join(dirname(__file__), '..', '..', '..', 'fftpack', 'tests')
+
+# Matlab reference data
+MDATA = np.load(join(fftpack_test_dir, 'test.npz'))
+X = [MDATA['x%d' % i] for i in range(8)]
+Y = [MDATA['y%d' % i] for i in range(8)]
+
+# FFTW reference data: the data are organized as follows:
+#    * SIZES is an array containing all available sizes
+#    * for every type (1, 2, 3, 4) and every size, the array dct_type_size
+#    contains the output of the DCT applied to the input np.linspace(0, size-1,
+#    size)
+FFTWDATA_DOUBLE = np.load(join(fftpack_test_dir, 'fftw_double_ref.npz'))
+FFTWDATA_SINGLE = np.load(join(fftpack_test_dir, 'fftw_single_ref.npz'))
+FFTWDATA_SIZES = FFTWDATA_DOUBLE['sizes']
```

Comment:
This pattern of "load at the top" is not so nice because it will actually cause the test discovery step of `pytest` to be slower.

A better pattern now that we use `pytest` is to make use of parametrizations. For example you could make one in this file with `module` scope that yielded the size and/or reference data on demand (and even implicitly allowed tests to iterate over it), when the test is actually run. For example if you wrote a fixture called `fftwdata_size` then the current code like this:
```
    def test_definition(self):
        for i in FFTWDATA_SIZES:
            ...
```
Would become:
```
    def test_definition(self, fftwdata_size):
        ...
```
But I don't think it's worth holding up this PR for it, especially since I assume the current pattern is modeled on existing code anyway.